src/components/Tooltip/TooltipPosition.js

```diff
@@ -0,0 +1,164 @@
+import React, { PureComponent, cloneElement } from 'react'
+import { findDOMNode }                        from 'react-dom'
+import PropTypes                              from 'prop-types'
+
+const styles = {
+  root: {
+    position: 'absolute',
+    zIndex: 99999,
+  },
+}
+
+const SPACING = 8
+const ARROW_SPACING = 6
+const ARROW_BORDER_SPACING = 2
+
+class TooltipPosition extends PureComponent {
+  static propTypes = {
+    children: PropTypes.node.isRequired,
+    target: PropTypes.oneOfType([
+      PropTypes.node,
+      PropTypes.func
+    ]).isRequired,
+    placement: PropTypes.oneOf([
+      'top',
+      'left',
+      'right',
+      'bottom',
+    ]).isRequired,
+  }
+
+  state = {
+    overlayRect: {},
+    arrowPosition: {}
+  }
+
+  componentDidMount() {
+    this.calculatePosition()
+  }
+
+  getTarget = () => {
+    const { target } = this.props
+    const targetNode = typeof target === 'function' ? target() : target
+    return targetNode && findDOMNode(targetNode) || null
```

Comment:
Is it possible to pass the node itself as the prop here so that we don't need to call `findDOMNode`?